isis/scripts/downloadIsisData.py

```diff
@@ -0,0 +1,201 @@
+             
+
+def download_pub(inputcommand, destination, mission_name, cfg, dry_run):
+    """
+    Parameters 
+    ----------
+    
+    destination str 
+            path to location where files will be copied/downloaded too
+    
+    set_of_pub : set(str)
+            set of kernels that will be downloaded
+    
+    cfg : str
+            config file
+    """
+    
+    log.info("Starting to download kernels")
+
+    if(inputcommand == "lsf"):
+        mission_name = mission_name.strip("/")
+        extra_args= [f"{mission_name}", "-R", "--format", "p", "--files-only"]
+        results = rclone(command="lsf", extra_args=extra_args, config=cfg, redirect_stdout=True, redirect_stderr=False)
+        f = open(f"{destination}/output.json" , "w")
+        f.truncate()
+        f.write(json.loads(json.dumps(results.get('out').decode('utf-8'))))
+        f.close()
+    else:
+        destination += "/"+str(mission_name).replace(":","")
+        extra_args=[f"{mission_name}",f"{destination}", "--progress", "--track-renames"]
+
+    if dry_run:
+        extra_args.append("--dry-run")
+
+
+    rclone(command=inputcommand, extra_args=extra_args, config=cfg, redirect_stdout=False, redirect_stderr=False)
+    log.info("Done downloading files")
+
+
+def main(command, mission, dest, legacy_flag, dry_run):
+    """
+    Parameters 
+    ----------
+    
+    source str 
+            path to location where files are being copied/downloaded from
+    
+    dest : str
+               path to location where files will be copied/downloaded too
+    """
+    log.info("---Script starting---")
+    
+    cfg_path = (os.path.dirname(__file__) + '/../config/rclone.conf')
+    
+    result = rclone("listremotes", config=cfg_path, )
+    results = result.get('out').decode("utf-8").split("\n")
+
+    if legacy_flag:
+        legacy_remotes = ["legacy_base_usgs"]
+        for remote in legacy_remotes:
+            driver_for_missions(command, remote, results, cfg_path, dest, dry_run)
+
+    if(mission.upper() == "ALL"):
+        mission_arr =  ["tgo", "dawn", "cassini", "hayabusa2", "juno", "odyssey", "mro", "mex", "cassis", "apollo", "base", "hayabusa", "chandrayaan1", "clementine1", "kaguya", "mariner10", "messenger", "mgs", "near", "newhorizons", "osirisrex", "rosetta", "smart1", "viking1", "viking2", "voyager1", "voyager2"]
+        for mission in mission_arr:
+
+            if f"{mission}_pub" in results:
+                driver_for_missions(command, f"{mission}_pub", results, cfg_path, dest, dry_run)
+
+            if f"{mission}_usgs" in results:
+                driver_for_missions(command, f"{mission}_usgs", results, cfg_path, dest, dry_run)
+
+    else:
+        if f"{mission}_pub" in results:
+            driver_for_missions(command, f"{mission}_pub", results, cfg_path, dest, dry_run)
+
+        if f"{mission}_usgs" in results:
+            driver_for_missions(command, f"{mission}_usgs", results, cfg_path, dest, dry_run)
```

Comment:
this `f"{mission}_pub"`/`f"{mission}_usgs"` need a colon at the end otherwise the strings won't match.